### PR TITLE
ballle98/AqualinkD#37: main process exited, code=exited, status=1

### LINKED ISSUE #88: heap-buffer-overflow in action_web_request
see ballle98/AqualinkD#37: main process exited, code=exited, status=1/FAILURE

```
08:13:21.215 Info:   URI request: '/api '
=================================================================
==14210==ERROR: AddressSanitizer: heap-buffer-overflow on address 0xb4403680 at pc 0x0003e84c bp 0xbe969674 sp 0xbe969240
READ of size 1025 at 0xb4403680 thread T0
    #0 0x3e84b in StrstrCheck(void*, char*, char const*, char const*) (/home/pi/remote-debugging/aqualinkd+0x3e84b)
    #1 0x3eab3 in strstr (/home/pi/remote-debugging/aqualinkd+0x3eab3)
    #2 0x113787 in action_web_request C:\git\AqualinkD/net_services.c:629
    #3 0x118abf in ev_handler C:\git\AqualinkD/net_services.c:1073
    #4 0x13adbb in mg_call C:\git\AqualinkD/mongoose.c:2258
    #5 0x15d5f7 in mg_http_call_endpoint_handler C:\git\AqualinkD/mongoose.c:8463
    #6 0x152557 in mg_http_handler C:\git\AqualinkD/mongoose.c:6374
    #7 0x13adbb in mg_call C:\git\AqualinkD/mongoose.c:2258
    #8 0x13f88b in mg_recv_common C:\git\AqualinkD/mongoose.c:2706
    #9 0x13f8db in mg_if_recv_tcp_cb C:\git\AqualinkD/mongoose.c:2710
    #10 0x14565f in mg_handle_tcp_read C:\git\AqualinkD/mongoose.c:3608
    #11 0x146277 in mg_mgr_handle_conn C:\git\AqualinkD/mongoose.c:3733
    #12 0x148007 in mg_socket_if_poll C:\git\AqualinkD/mongoose.c:3925
    #13 0x13cfeb in mg_mgr_poll C:\git\AqualinkD/mongoose.c:2424
    #14 0xf9def in main_loop C:\git\AqualinkD/aqualinkd.c:1236
    #15 0xf811b in main C:\git\AqualinkD/aqualinkd.c:808
    #16 0xb6d03677 in __libc_start_main (/lib/arm-linux-gnueabihf/libc.so.6+0x16677)

0xb4403680 is located 0 bytes to the right of 1024-byte region [0xb4403280,0xb4403680)
allocated by thread T0 here:
    #0 0xbe557 in malloc (/home/pi/remote-debugging/aqualinkd+0xbe557)
    #1 0x145503 in mg_handle_tcp_read C:\git\AqualinkD/mongoose.c:3572
    #2 0x146277 in mg_mgr_handle_conn C:\git\AqualinkD/mongoose.c:3733
    #3 0x148007 in mg_socket_if_poll C:\git\AqualinkD/mongoose.c:3925
    #4 0x13cfeb in mg_mgr_poll C:\git\AqualinkD/mongoose.c:2424
    #5 0xf9def in main_loop C:\git\AqualinkD/aqualinkd.c:1236
    #6 0xf811b in main C:\git\AqualinkD/aqualinkd.c:808
    #7 0xb6d03677 in __libc_start_main (/lib/arm-linux-gnueabihf/libc.so.6+0x16677)

SUMMARY: AddressSanitizer: heap-buffer-overflow (/home/pi/remote-debugging/aqualinkd+0x3e84b) in StrstrCheck(void*, char*, char const*, char const*)
Shadow bytes around the buggy address:
  0x36880680: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x36880690: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x368806a0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x368806b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x368806c0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x368806d0:[fa]fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x368806e0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa


```

### PR DESCRIPTION
**Pull request by ballle98**
fix #88

### CODE REVIEW

File: aqualinkd.c [hunk 1]
```diff
@@ -694,9 +694,7 @@ int main(int argc, char *argv[])
   bool cmdln_debugRS485 = false;
 
   // struct lws_context_creation_info info;
-  // Log only NOTICE messages and above. Debug and info messages
-  // will not be logged to syslog.
-  setlogmask(LOG_UPTO(LOG_NOTICE));
+  setlogmask(LOG_UPTO(LOG_DEBUG));
```
Comment (ballle98) [inline aqualinkd.c]: When the mask is set to NOTICE if the user enables DEBUG in aqualinkd.conf journaclt will only show the DEBUG logs.  all the other ones are lost.  Opposite the behavior you'd expect

**Comment by sfeakes** [comment]
This is by design.   Don’t want debug info in syslog.   That’s why the log to file option is in the config, and that does follow debug level.

**Comment by ballle98** [comment]
Understood.  I was just saying what actually happens with current master.  If you run aqualinkd as a service and you set DEBUG as the logging level in /etc/adualinkd.conf then look at the logs that show up using "journalctl -u aqualinkd" or systemctl status aqualinkd you will see the DEBUG logs and only DEBUG logs, seems like a bug in journald.  I can take this change out but you should add a check n main() if your a daemon and the log level is greater than NOTICE force it to NOTICE. That said I'm running on a crummy B+ Pi and DEBUG logs work fine, they don't break the damon, and were useful in debugging this issue.

> This is by design. Don’t want debug info in syslog. That’s why the log to file option is in the config, and that does follow debug level.

**Comment by sfeakes** [comment]
Interesting about only seeing DEBUG logs with journalctl, I'll have to look into that.  Main reason I didn't want DEBUG to syslog is for most users getting them to post just the aqualinkd log file is hard enough, let alone asking to grabbing them from syslog. For developing this I agree, syslog is better.

File: aqualinkd.c (hunk 1 above)
Comment (ballle98) [reply aqualinkd.c]: I remove all changes related to syslog and just kept the fix for #88 heap-buffer-overflow in action_web_request

**Comment by ballle98** [comment]
I remove all changes related to syslog and just kept the fix for #88 heap-buffer-overflow in action_web_request

**Pull request closed without merging**